### PR TITLE
[EC-318] DevOps Pipelines migration to GitHub Actions

### PR DESCRIPTION
**Pull request by mamu0**
### List of changes

Added GitHub Actions for

- core code review (Legacy and new mode)
- core deploy (Legacy and new mode)
- backup apim deploy

All of this pipe need to be created for testing purposse.

### Motivation and context
We want to bring all the pipelines we use on Azure DevOps to GitHub to use Actions.

### Type of changes

- [x] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Env to apply

- [ ] DEV
- [ ] UAT
- [x] PROD

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [x] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [x] No

### Other information

---

### How to apply

After PR is approved
1. run deploy pipeline from Azure DevOps [io-platform-iac-projects](https://dev.azure.com/pagopaspa/io-platform-iac-projects/_build?view=folders)
2. select PR branch
3. wait for approval

### CODE REVIEW

File: .github/workflows/core_code_review_weu.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    types:
+      - opened
+      - edited
+      - synchronize
+      - reopened
+      - ready_for_review
+    paths:
+      - "src/core/*"
+      - "src/psql/*"
+      - ".devops/core-code-review-pipelines.yml"
```
Comment (christian-calabrese) [inline .github/workflows/core_code_review_weu.yaml:15]: If you agree, I would change it with the path of this action to trigger it on change
```suggestion
      - ".github/workflows/core_code_review_weu.yaml"
```

File: .github/workflows/core_deploy_weu.yaml [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        required: true
+        default: 'true'
+        options:
+          - true
+          - false
+      use_private_agent:
+        description: Use a private agent to run the Terraform plan.
+        type: boolean
+        required: false
+        default: false
```
Comment (christian-calabrese) [inline .github/workflows/core_deploy_weu.yaml:19]: To check if the self-hosted runner is mandatory due to the existence of private resources

**Review by christian-calabrese — CHANGES_REQUESTED** [review]

File: .github/workflows/core_code_review_weu.yaml (hunk 1 above)
Comment (mamu0) [reply .github/workflows/core_code_review_weu.yaml:15]: I agree

**Review by christian-calabrese — APPROVED** [review]
LGTM

**Pull request merged**